# Tarea No.1

## Computación

## Física, UNAM.

Castro Granados Celeste. 


## Sistema de números de punto flotante

**Características:** 

1. Está basado en la notación científica
2. Capaz de representar números muy grandes y muy pequeños sin incrementar el número de bits.
3. Capaz de representar números con componentes enteros y fraccionarios.


>Número de Punto Floante = Número Real.

**Partes de un número de punto flotante:** 

* Mantisa: Magnitud del número.
* Exponente: Número de lugares que se va a mover el punto.
* Signo: Positivo o negativo.

Ejemplo:

In [2]:
Número=241,506,800
Mantisa=.2415068

0.2415068

In [3]:
Exponente=9

9

Lo anterior se refiere a : 0.2415068 x10^9

Aquí podemos ver otro ejemplo de número flotante:

In [9]:
typeof(2.6789)

Float64

Para los números de punto flotante binarios el formato se define por el standard ANSI/IEEE 754-1985 de tres formas:

+ Precisión sencilla - 32 bits
+ Precisión doble - 64 bits
+ Precisión extendida - 80 bits
+ Se trabaja con números normalizados.

>Nota: Decimos que un número binario está normalizado si el dígito a la izquierda del punto es igual a 1.

![Imágen](http://slideplayer.es/slide/5388201/17/images/25/Partes+de+un+n%C3%BAmero+de+punto+flotante.jpg)

## Diferencia entre sistemas operativos de 32 bits y 64 bits

### Procesadores:

Los procesadores de 64 bits pueden trabajar el doble de información en el mismo ciclo de reloj (un hertz), pueden acceder a mayor capacidad de memoria y procesar archivos más grandes. Actualmente un procesador de 32 bits puede controlar un máximo de 4 GB de memoria Ram; mientras que un procesador de 64 bits tiene la capacidad de controlar 16 Exabytes de memoria, es decir, 16 mil millones de GB, una cantidad bastante sorprendente.

En cuanto a los cálculos matemáticos también existen diferencias, ya que un procesador de 32 bits puede representar números desde 0 hasta 4,294,967,295; y uno de 64 bits incrementará la capacidad logrando que se puedan representar números desde 0 hasta 18,446744,073,709,551,615. Obviamente esto significa que las computadoras podrán hacer operaciones con cantidades mayores y que los cálculos con cantidades pequeñas sean más eficientes.

Las diferencias son bastante notables y sin duda contar con un procesador de 64 bits mejoraría sin duda las capacidades de hardware, en especial el poder trabajar con bastante memoria ram.

### Sistemas operativos:

Cuando encontramos diferentes versiones de algún sistema operativo (Windows, Linux o Mac) y debemos elegir entre una versión de 32 o una de 64 bits debemos tener en cuenta lo siguiente:

1. Para instalar una versión de 64 bits en nuestro computador debemos contar con un procesador de 64 bits o de lo contrario no podremos instalarlo.
2. Las versiones de 32 bits se puede instalar con cualquier tipo de procesador ya sea de 32 o 64 bits
3. Si instalamos una versión de 32 bits en un computador con procesador de 64 estaremos desaprovechando las ventajas que dicho procesador trae, ya que para sacarle todo su potencial es necesario tener un sistema de 64 bits instalado.
4. Los sistemas operativos de 64 bits son mas utilizados a nivel profesión donde se requiere realmente todo el potencial de la maquina, ademas muchos sistemas de 64 bits presentan incompatibilidad gran cantidad de software.


### Software:

Cuando debemos escoger entre software de 32 y 64 bits lo primero y mas importante a tener en cuenta es que tipo de Sistema Operativo tenemos instalado y si es de 32 o 64 bits para así respectivamente seleccionar el software a instalar.

Muchos sistemas de 64 bits presentan incompatibilidad al instalarle software de 32 bits, por eso se recomienda instalarle solo versiones de 64 bits para un mejor rendimiento y evitar problemas, aunque los sistemas actuales ya vienen con muchas mejoras para evitar esto.


![Imágen2](http://omicrono.elespanol.com/wp-content/uploads/2016/08/32-bit-y-64-bit.jpg)

## ALU (Unidad aritmética lógica)

La unidad aritmético lógica, también conocida como ALU (siglas en inglés de arithmetic logic unit), es un circuito digital que calcula operaciones aritméticas y operaciones lógicas, entre dos números.

*Operaciones típicas:* 
1. Sumar
2. Restar
3. Multiplicar
4. Desplazamiento de registros
5. Comparaciones 

>Ejemplos de operaciones:

In [10]:
34+789

823

In [11]:
4563-9047

-4484

In [12]:
2121*456

967176

![Imágen3](http://seduca.uaemex.mx/material/LIA/LC/img/lc_U4t4_1.jpg)

## Algoritmo y Estabilidad

**Nociones de estabilidad**

Sea *X* un conjunto de datos de entrada, *Y* el conjunto de datos de salida y *f* la forma en la que se relacionan *X* y *Y*.
Entonces: *f*: *X* ---> *Y*.
Sin embargo para poder encontrar una solución es necesario aproximarla mediante algún algoritmo computacional.

Debido a que en la mayoría de los casos los algoritmos empleados están escritos en algún lenguaje de computación, los datos de entrada (subconjunto de *X*), así como las operaciones tienen un error, ya sea por el redondeo o por los cálculos realizados.
Es por este motivo que lo más que se le puede pedir a un algoritmo numérico es que sea estable. 

>Idea intuitiva: Un algoritmo es establesi, pequeños cambios en los datos de entrada, producen pequeños cambios en los datos de salida. Y por el contrario, un algoritmo es inestable si, pequeños cambios en los datos de entrada producen grandes cambios en los datos de salida. 

Un algoritmo es estable para una clase de matrices C si para cada matriz A en , la solución computacional por el algoritmo es la solución exacta cercana al problema. 

Un problema (con respecto a un conjunto dado de datos) es llamado mal condicionado si a una perturbación relativa pequeña en los datos puede causar un error rlativo grande en la solución computacional, independientemente del método de solución. De lo contrario, es llamado bien condicionado; esto es, un problema es bien condicionao si todas las pequeñas perturbaciones en los datos sólo producen pequeños errores relativos en la solución.

## Cibergrafía

1. <http://www.erikavilches.com/Anterior/TC1004.01.200811/diapositivas/Punto%20Flotante%201.pdf>
2. <https://www.maestrodelacomputacion.net/diferencia-entre-32-y-64-bits/>
3. <http://profesores.fi-b.unam.mx/normaelva/ALU.pdf>
4. <https://prezi.com/untsczyf_gst/estabilidad-de-algoritmos/>
